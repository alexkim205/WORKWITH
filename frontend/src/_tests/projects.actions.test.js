import createMockStore from '../_config/mockStore.config';
import {
  getProjectsByUser,
  getProjects,
  getProject,
  createProject,
  updateProject,
  deleteProject
} from '../_actions/projects.actions';
import testConstants from '../_constants/test.constants';

const {
  users: { user },
  projects: { project, newProject, newUpdateProject }
} = testConstants;

describe('Project Actions', () => {
  let store;
  beforeAll(() => {
    store = createMockStore();
  });
  afterEach(() => {
    store.clearActions();
  });
  it('it should GET projects by user', async () => {
    await store.dispatch(getProjectsByUser(user._id));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should GET projects', async () => {
    await store.dispatch(getProjects());
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should GET a project', async () => {
    await store.dispatch(getProject(project._id));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should CREATE a project', async () => {
    await store.dispatch(createProject(newProject));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should UPDATE a project', async () => {
    await store.dispatch(updateProject(project._id, newUpdateProject));
    expect(store.getActions()).toMatchSnapshot();
  });
  it('it should DELETE a project', async () => {
    await store.dispatch(deleteProject(project._id));
    expect(store.getActions()).toMatchSnapshot();
  });
});
