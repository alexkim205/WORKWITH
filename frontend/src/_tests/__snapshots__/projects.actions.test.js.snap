// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Actions it should CREATE a project 1`] = `
Array [
  Object {
    "type": "projects/create-project-pending",
  },
  Object {
    "error": null,
    "payload": Object {
      "__v": 0,
      "_id": "5e02efdd94748b21482e6010",
      "authors": Array [
        "5df9778c0f65fddc651e6167",
      ],
      "createdAt": "2019-12-25T05:13:01.264Z",
      "deleted": false,
      "private": true,
      "title": "My Project",
      "updatedAt": "2019-12-25T05:31:27.794Z",
      "users": Array [
        "5df9778c0f65fddc651e6167",
        "5df97791556bc5bc928b32fb",
      ],
    },
    "type": "projects/create-project-success",
  },
]
`;

exports[`Project Actions it should DELETE a project 1`] = `
Array [
  Object {
    "type": "projects/delete-project-pending",
  },
  Object {
    "error": null,
    "payload": undefined,
    "type": "projects/delete-project-success",
  },
]
`;

exports[`Project Actions it should GET a project 1`] = `
Array [
  Object {
    "type": "projects/get-project-pending",
  },
  Object {
    "error": null,
    "payload": Object {
      "__v": 0,
      "_id": "5e02efdd94748b21482e6010",
      "authors": Array [
        "5df9778c0f65fddc651e6167",
      ],
      "createdAt": "2019-12-25T05:13:01.264Z",
      "deleted": false,
      "private": true,
      "title": "My Project",
      "updatedAt": "2019-12-25T05:31:27.794Z",
      "users": Array [
        "5df9778c0f65fddc651e6167",
        "5df97791556bc5bc928b32fb",
      ],
    },
    "type": "projects/get-project-success",
  },
]
`;

exports[`Project Actions it should GET projects 1`] = `
Array [
  Object {
    "type": "projects/get-projects-pending",
  },
  Object {
    "error": null,
    "payload": Array [
      Object {
        "__v": 0,
        "_id": "5e02efdd94748b21482e6010",
        "authors": Array [
          "5e0bb5c61c90231c34596f83",
        ],
        "createdAt": "2019-12-25T05:13:01.264Z",
        "deleted": false,
        "private": true,
        "title": "Hera",
        "updatedAt": "2019-12-25T05:31:27.794Z",
        "users": Array [
          "5e0bb5c61c90231c34596f83",
          "5df9778c0f65fddc651e6167",
          "5df97791556bc5bc928b32fb",
        ],
      },
      Object {
        "__v": 0,
        "_id": "5e4c34e5890521be2c2de2fe",
        "authors": Array [
          "5e0bb5c61c90231c34596f83",
        ],
        "createdAt": "2019-12-25T05:13:01.264Z",
        "deleted": false,
        "private": true,
        "title": "Poseidon",
        "updatedAt": "2019-12-26T05:31:27.794Z",
        "users": Array [
          "5e0bb5c61c90231c34596f83",
          "5df9778c0f65fddc651e6167",
          "5df97791556bc5bc928b32fb",
        ],
      },
      Object {
        "__v": 0,
        "_id": "5e4c34e91ff8893429ff8ce6",
        "authors": Array [
          "5e0bb5c61c90231c34596f83",
        ],
        "createdAt": "2019-12-25T05:13:01.264Z",
        "deleted": false,
        "private": true,
        "title": "Hades",
        "updatedAt": "2019-12-27T05:31:27.794Z",
        "users": Array [
          "5e0bb5c61c90231c34596f83",
          "5df9778c0f65fddc651e6167",
          "5df97791556bc5bc928b32fb",
        ],
      },
      Object {
        "__v": 0,
        "_id": "5e4c34f192e646c363c3a803",
        "authors": Array [
          "5e0bb5c61c90231c34596f83",
        ],
        "createdAt": "2019-12-25T05:13:01.264Z",
        "deleted": false,
        "private": true,
        "title": "Dionysus",
        "updatedAt": "2019-12-28T05:31:27.794Z",
        "users": Array [
          "5e0bb5c61c90231c34596f83",
          "5df9778c0f65fddc651e6167",
          "5df97791556bc5bc928b32fb",
        ],
      },
    ],
    "type": "projects/get-projects-success",
  },
]
`;

exports[`Project Actions it should GET projects by user 1`] = `
Array [
  Object {
    "type": "projects/get-user-projects-pending",
  },
  Object {
    "error": null,
    "payload": Array [
      Object {
        "__v": 0,
        "_id": "5e02efdd94748b21482e6010",
        "authors": Array [
          "5e0bb5c61c90231c34596f83",
        ],
        "createdAt": "2019-12-25T05:13:01.264Z",
        "deleted": false,
        "private": true,
        "title": "Hera",
        "updatedAt": "2019-12-25T05:31:27.794Z",
        "users": Array [
          "5e0bb5c61c90231c34596f83",
          "5df9778c0f65fddc651e6167",
          "5df97791556bc5bc928b32fb",
        ],
      },
      Object {
        "__v": 0,
        "_id": "5e4c34e5890521be2c2de2fe",
        "authors": Array [
          "5e0bb5c61c90231c34596f83",
        ],
        "createdAt": "2019-12-25T05:13:01.264Z",
        "deleted": false,
        "private": true,
        "title": "Poseidon",
        "updatedAt": "2019-12-26T05:31:27.794Z",
        "users": Array [
          "5e0bb5c61c90231c34596f83",
          "5df9778c0f65fddc651e6167",
          "5df97791556bc5bc928b32fb",
        ],
      },
      Object {
        "__v": 0,
        "_id": "5e4c34e91ff8893429ff8ce6",
        "authors": Array [
          "5e0bb5c61c90231c34596f83",
        ],
        "createdAt": "2019-12-25T05:13:01.264Z",
        "deleted": false,
        "private": true,
        "title": "Hades",
        "updatedAt": "2019-12-27T05:31:27.794Z",
        "users": Array [
          "5e0bb5c61c90231c34596f83",
          "5df9778c0f65fddc651e6167",
          "5df97791556bc5bc928b32fb",
        ],
      },
      Object {
        "__v": 0,
        "_id": "5e4c34f192e646c363c3a803",
        "authors": Array [
          "5e0bb5c61c90231c34596f83",
        ],
        "createdAt": "2019-12-25T05:13:01.264Z",
        "deleted": false,
        "private": true,
        "title": "Dionysus",
        "updatedAt": "2019-12-28T05:31:27.794Z",
        "users": Array [
          "5e0bb5c61c90231c34596f83",
          "5df9778c0f65fddc651e6167",
          "5df97791556bc5bc928b32fb",
        ],
      },
    ],
    "type": "projects/get-user-projects-success",
  },
]
`;

exports[`Project Actions it should UPDATE a project 1`] = `
Array [
  Object {
    "type": "projects/update-project-pending",
  },
  Object {
    "error": null,
    "payload": Object {
      "__v": 0,
      "_id": "5e02efdd94748b21482e6010",
      "authors": Array [
        "5df9778c0f65fddc651e6167",
      ],
      "createdAt": "2019-12-25T05:13:01.264Z",
      "deleted": false,
      "private": true,
      "title": "My Project",
      "updatedAt": "2019-12-25T05:31:27.794Z",
      "users": Array [
        "5df9778c0f65fddc651e6167",
        "5df97791556bc5bc928b32fb",
      ],
    },
    "type": "projects/update-project-success",
  },
]
`;
